File: BackEnd/DotNet/WebDigital/WebDigital/Services/Implements/AuthorService.cs
using Microsoft.EntityFrameworkCore;
using WebDigital.dto;
using WebDigital.Models;
using WebDigital.Services.Interface;

namespace WebDigital.Services.Implements
{
    public class AuthorService : IAuthorService
    {
        private readonly BookwormfinalBackupContext _context;

        public AuthorService(BookwormfinalBackupContext context)
        {
            _context = context;
        }

        public async Task<List<AuthorDto>> GetAll()
        {
            return await _context.Authors
                .OrderBy(a => a.Name)
                .Select(a => new AuthorDto
                {
                    AuthorId = a.AuthorId,
                    Name = a.Name,
                    Bio = a.Bio
                })
                .ToListAsync();
        }

        public async Task<int> Create(AuthorCreateDto dto)
        {
            var author = new Author
            {
                Name = dto.Name,
                Bio = dto.Bio
            };

            _context.Authors.Add(author);
            await _context.SaveChangesAsync();

            return author.AuthorId;
        }
    }
}

File: BackEnd/Java/product-service/src/main/java/com/digitalshelf/productservice/service/AuthorService.java
package com.digitalshelf.productservice.service;

import com.digitalshelf.productservice.dto.AuthorCreateDto;
import com.digitalshelf.productservice.dto.AuthorDto;
import com.digitalshelf.productservice.entity.Author;
import com.digitalshelf.productservice.repository.AuthorRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Comparator;
import java.util.List;

@Service
@RequiredArgsConstructor
public class AuthorService {

    private final AuthorRepository authorRepository;

    public List<AuthorDto> getAll() {
        return authorRepository.findAll().stream()
                .sorted(Comparator.comparing(Author::getName))
                .map(a -> AuthorDto.builder().authorId(a.getAuthorId()).name(a.getName()).bio(a.getBio()).build())
                .toList();
    }

    public Integer create(AuthorCreateDto dto) {
        Author author = new Author();
        author.setName(dto.getName());
        author.setBio(dto.getBio());
        authorRepository.save(author);
        return author.getAuthorId();
    }
}